docs/concepts/state-management.md

```diff
@@ -0,0 +1,33 @@
+---
+title: State Management
+description: An overview of state management in FlutterFlow
+---
+
+State management is a crucial concept focused on maintaining and controlling the *state* of an application. Simply put, it involves monitoring the changes within your app and updating the user interface to reflect these changes. 
+
+You can think of the UI as a function of your state variables, where changes in these variables directly influence what the UI displays.
+
+In FlutterFlow, there are a few types of state variables that you can create:
+
+
+![App State is shared across multiple pages in the application. Component State is specific to a component. Page State is shared across widgets on the page.](<state_management_overview.png>)
```

Comment:
The alt text does not show up on the UI actually. Can we add a captioned image with that text as well? 

Please [check point 16 in Writing Guidelines doc](https://www.notion.so/flutterflow/Docs-Contribution-Guidelines-42829c305df441f6b061b4095abac8a8) for how to add one!